File: api-spring/src/main/java/com/refri/api/user/dto/UpdateUserDto.java
package com.refri.api.user.dto;

import com.refri.api.user.type.Diet;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.URL;

@Data
@AllArgsConstructor
@NoArgsConstructor()
public class UpdateUserDto {
    @NotBlank
    private String username;

    @URL
    private String thumbnail;

    @NotBlank
    private String introduction;

    private Diet diet;
}

File: api/libs/user/src/dto/update-user.dto.ts
import { IsEnum, IsOptional, IsString, IsUrl } from 'class-validator';
import { Diet } from '@app/user/domain/diet.enum';

export class UpdateUserDto {
  @IsOptional()
  @IsString()
  username?: string;

  @IsOptional()
  @IsUrl()
  thumbnail?: string;

  @IsOptional()
  @IsString()
  introduction?: string;

  @IsOptional()
  @IsEnum(Diet)
  diet?: Diet;
}
